### PR TITLE
Test publishing

### PR DESCRIPTION
**Pull request by breedx-splk**
Ok, this is purely for helping along the process.  The Dockerfile shouldn't exist, it just make repeating some things slightly more bearable.

Some notes:

## Env setup

Assuming you have already created a signing key and know the ID of it (`gpg --list-key`), you need to set up the key and password in the environment.

```
$ export GPG_SECRET_KEY=$(gpg --armor --export-secret-key <your-key-id>)
$ IFS= read -rsp 'Please the key password: ' GPG_PASSWORD
```
now you have those in your local environment.  

## Build container

```
$ docker build .
```

Yup, that's it. Make a note of the container ID listed at the bottom of the output.

## Running the signing task

Don't expand these vars, just let your shell do it:

```
docker run -it --rm \
    -e ORG_GRADLE_PROJECT_signingKey="$GPG_SECRET_KEY" \
    -e ORG_GRADLE_PROJECT_signingPassword=${GPG_PASSWORD}
```

#### try more stuff

To try more things, interactively, you can add `/bin/bash` on the end of the above docker command. Once you get that shell, run `./do_signing.sh`

# What's not working

So the signing task wants to sign publication _components_. Right now, the maven publication doesn't seem to know about the android _components_ or what it produces. Until that is set up (I have no idea how to do that), the signing won't work. See the `publications` block in the `publishing` block in the `splunk-otel-android` submodule.

Additional docs: https://docs.gradle.org/current/userguide/publishing_customization.html#sec:adding-variants-to-existing-components

Additional docs also: https://docs.gradle.org/current/userguide/signing_plugin.html

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,20 +6,25 @@ buildscript {
...
-        classpath 'com.vanniktech:gradle-maven-publish-plugin:0.16.0'
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
     }
 }
 
+plugins {
+  id 'signing'
+  id 'maven-publish'
```
Comment (jkwatson) [inline build.gradle:17]: is this needed here? even if we're only publishing/signing the splunk-otel-java module?

File: build.gradle [hunk 2]
```diff
@@ -6,20 +6,25 @@ buildscript {
...
 allprojects {
     repositories {
         google()
         mavenCentral()
     }
+
 }
 
-task clean(type: Delete) {
+task cleanx(type: Delete) {
```
Comment (jkwatson) [inline build.gradle:28]: ?

File: splunk-otel-android/build.gradle [hunk 3]
```diff
@@ -64,4 +65,34 @@ dependencies {
...
+publishing {
+  repositories {
+    maven {
+      name = "GitHubPackages"
+      url = uri("https://maven.pkg.github.com/signalfx/splunk-otel-android")
+      credentials {
+        username = System.getenv("GITHUB_ACTOR_FIXMEXXX")
+        password = System.getenv("GITHUB_TOKEN_FIXMEXXX")
+      }
+    }
```
Comment (jkwatson) [inline splunk-otel-android/build.gradle:79]: why did you add this bit?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+RUN ANDROID_SDK_ROOT=/app ./gradlew -x test build
+
+RUN echo 'export ANDROID_SDK_ROOT=/app' > do_signing.sh
+# RUN echo './gradlew -x test -PsigningKeyId=${GPG_KEY_ID} -PsigningInMemoryKey=$GPG_SECRET_KEY -PsigningInMemoryKeyPassword=$GPG_PASSWORD' signMavenPublication >> ./do_signing.sh
+
+RUN echo './gradlew -x test signMavenPublication' >> ./do_signing.sh
+RUN chmod 755 /app/splunk-otel-android/do_signing.sh
+
+
+CMD './do_signing.sh'
```
Comment (jkwatson) [inline Dockerfile:42]: curious about this rigamarole. Any reason why we can't just directly call gradle here?

File: build.gradle (hunk 2 above)
Comment (jkwatson) [reply build.gradle:28]: I got it. there's already a `clean` task from one of the plugins.

File: build.gradle (hunk 1 above)
Comment (breedx-splk) [reply build.gradle:17]: probably not

File: build.gradle (hunk 2 above)
Comment (breedx-splk) [reply build.gradle:28]: had to disable that because a `clean` task already exists.

File: splunk-otel-android/build.gradle (hunk 3 above)
Comment (breedx-splk) [reply splunk-otel-android/build.gradle:79]: copypasta!!

File: Dockerfile (hunk 4 above)
Comment (breedx-splk) [reply Dockerfile:42]: probably can. the script just allowed me to try some other things, like echoing environment vars etc. to make sure things were set up properly.

**Comment by breedx-splk** [comment]
It's publishing! 🎉
